.github/PULL_REQUEST_TEMPLATE.md

```diff
@@ -0,0 +1,25 @@
+Thank you for submitting the pull request to the Apache Ignite.
+
+In order to streamline the review of the contribution 
+we ask you to ensure the following steps have been taken:
+
+### The Contribution Checklist
+- [ ] The JIRA ticket has been associated with the pull request
+- [ ] The JIRA ticket has the _Patch Available_ state
+- [ ] The pull request briefly describes changes that have been made
+- [ ] The pull request title has the following pattern 
+e.g. `IGNITE-12407: Java thin client. Cluster API support`
+- [ ] Commits have the following pattern 
+e.g. `IGNITE-12835: Thin client. Compute support added`
+- [ ] A reviewer has been asked for the review through the JIRA comments 
+(see [the Maintainers list](https://cwiki.apache.org/confluence/display/IGNITE/How+to+Contribute#HowtoContribute-ReviewProcessandMaintainers)) 
+- [ ] The pull request has been checked by the Teamcity Bot and 
+the `green visa` applied to the JIRA ticket (see [TC.Bot: Check PR](https://mtcga.gridgain.com/prs.html))
```

Comment:
Shall we consider using travis build results for PR requests? Since travis build is already part of PR request process and build results are already available in the PR.

Reply:
Currently, travisci is just a helpful check for PR and the `greenvisa` still required. For most PRs it is necessary to have all CI passed successfully.